test/suite/test_prepare_hs03.py

```diff
@@ -0,0 +1,211 @@
+        self.session.salvage(self.uri, "force")
+        self.session.verify(self.uri, None)
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def prepare_updates(self, ds, nrows, nsessions, nkeys):
+        # Insert some records with commit timestamp, corrupt file and call salvage, verify before checkpoint.
+
+        # Commit some updates to get eviction and history store fired up
+        commit_value = b"bbbbb" * 100
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, nsessions * nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            cursor.set_key(ds.key(nrows + i))
+            cursor.set_value(commit_value)
+            self.assertEquals(cursor.insert(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(1))
+        cursor.close()
+
+        # Corrupt the table, Call salvage to recover data from the corrupted table and call verify
+        self.corrupt_salvage_verify()
+
+        # Call checkpoint
+        self.session.checkpoint()
+
+        hs_writes_start = self.get_stat(stat.conn.cache_write_hs)
+
+        # Have prepared updates in multiple sessions. This should ensure writing
+        # prepared updates to the history store
+        sessions = [0] * nsessions
+        cursors = [0] * nsessions
+        prepare_value = b"ccccc" * 100
+        for j in range (0, nsessions):
+            sessions[j] = self.conn.open_session()
+            sessions[j].begin_transaction('isolation=snapshot')
+            cursors[j] = sessions[j].open_cursor(self.uri)
+            # Each session will update many consecutive keys.
+            start = (j * nkeys)
+            end = start + nkeys
+            for i in range(start, end):
+                cursors[j].set_key(ds.key(nrows + i))
+                cursors[j].set_value(prepare_value)
+                self.assertEquals(cursors[j].insert(), 0)
+            sessions[j].prepare_transaction('prepare_timestamp=' + timestamp_str(4))
+
+        hs_writes = self.get_stat(stat.conn.cache_write_hs) - hs_writes_start
+
+        # Assert if not writing anything to the history store.
+        self.assertGreaterEqual(hs_writes, 0)
+
+        # Test if we can read prepared updates from the history store.
+        cursor = self.session.open_cursor(self.uri)
+        self.session.begin_transaction('read_timestamp=' + timestamp_str(1))
+        for i in range(1, nsessions * nkeys):
+            cursor.set_key(ds.key(nrows + i))
+            # The search should pass.
+            self.assertEqual(cursor.search(), 0)
+            # Correctness Test - commit_value should be visible
+            self.assertEquals(cursor.get_value(), commit_value)
+            # Correctness Test - prepare_value should NOT be visible
+            self.assertNotEquals(cursor.get_value(), prepare_value)
```

Comment:
This is redundant code, there's no way the previous assert can pass and this one will fail.

Reply:
I do agree with your concern here, but this code is to confirm that prepare_value is not visible after prepare_transactions and before commit.